tests/unit/amo/components/TestCollection.js

```diff
@@ -855,6 +915,23 @@ describe(__filename, () => {
     expect(manager).toHaveProp('collection', null);
   });
 
+  it('renders AuthenticateButton when creating and not signed in', () => {
+    const { store } = dispatchClientMetadata();
+    const location = fakeRouterLocation({
+      pathname: '/create/url/',
+    });
+    const root = renderComponent({ store, creating: true, location });
+
+    const authButton = root.find(AuthenticateButton);
+    expect(authButton).toHaveProp('location', location);
```

Comment:
we could maybe have an expectation for the button text, since it can be different.